parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeospatialStatistics.java

```diff
@@ -0,0 +1,260 @@
+      return new GeospatialStatistics.Builder();
+    } else if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.GeographyLogicalTypeAnnotation) {
+      EdgeInterpolationAlgorithm edgeAlgorithm =
+          ((LogicalTypeAnnotation.GeographyLogicalTypeAnnotation) logicalTypeAnnotation).getAlgorithm();
+      return new GeospatialStatistics.Builder(edgeAlgorithm);
+    } else {
+      return noopBuilder();
+    }
+  }
+
+  /**
+   * Constructs a GeospatialStatistics object with the specified CRS, bounding box, and geospatial types.
+   *
+   * @param boundingBox the bounding box for the geospatial data, or null if not applicable, note that
+   *    - The bounding box (bbox) is omitted only if there are no X or Y values.
+   *    - The Z and/or M statistics are omitted only if there are no Z and/or M values, respectively.
+   * @param geospatialTypes the geospatial types
+   */
+  public GeospatialStatistics(
+      BoundingBox boundingBox, GeospatialTypes geospatialTypes, EdgeInterpolationAlgorithm edgeAlgorithm) {
+    this.boundingBox = boundingBox;
+    this.geospatialTypes = geospatialTypes;
+    this.edgeAlgorithm = edgeAlgorithm;
+  }
+
+  /**
+   * Constructs a GeospatialStatistics object with the specified CRS.
+   */
+  public GeospatialStatistics() {
+    this(new BoundingBox(), new GeospatialTypes(), null);
+  }
+
+  /**
+   * Constructs a GeospatialStatistics object with the specified CRS and edge interpolation algorithm.
+   *
+   * @param crs the coordinate reference system
+   * @param edgeAlgorithm the edge interpolation algorithm
+   */
+  public GeospatialStatistics(String crs, EdgeInterpolationAlgorithm edgeAlgorithm) {
+    this.boundingBox = new BoundingBox();
+    this.geospatialTypes = new GeospatialTypes();
+    this.edgeAlgorithm = edgeAlgorithm;
+  }
+
+  /** Returns the bounding box. */
+  public BoundingBox getBoundingBox() {
+    return boundingBox;
+  }
+
+  /** Returns the geometry types. */
+  public GeospatialTypes getGeospatialTypes() {
+    return geospatialTypes;
+  }
+
+  /**
+   * @return whether the statistics has valid value.
+   */
+  public boolean isValid() {
+    if (boundingBox == null || geospatialTypes == null) {
+      return false;
+    }
+    return boundingBox.isValid() && geospatialTypes.isValid();
+  }
+
+  public void merge(GeospatialStatistics other) {
+    Preconditions.checkArgument(other != null, "Cannot merge with null GeometryStatistics");
+
+    // If other bounding box is null or invalid, set this as null
+    if (boundingBox == null || other.boundingBox == null) {
+      boundingBox = null;
+    }
+
+    if (boundingBox != null) {
+      boundingBox.merge(other.boundingBox);
+    }
+
+    // If other geosptialTypes is null or invalid, set this as null
+    if (geospatialTypes == null || other.geospatialTypes == null) {
+      geospatialTypes = null;
+    }
+
+    if (geospatialTypes != null) {
+      geospatialTypes.merge(other.geospatialTypes);
+    }
```

Comment:
```suggestion
    if (boundingBox != null) {
      boundingBox.merge(other.boundingBox);
    }
    if (geospatialTypes != null) {
      geospatialTypes.merge(other.geospatialTypes);
    }
```

`boundingBox` and `geospatialTypes` can manage the validity internally.